source/lubeck.d

```diff
@@ -1960,6 +1960,93 @@ struct QRResult(T)
     {
         return qrSolve!(allowDestroy)(matrix, tau, b);
     }
+    
+    /++
+    Extract the Q matrix
+    +/
+    auto Q(Flag!"allowDestroy" allowDestroy = No.allowDestroy)
+    {
+        auto work = [matrix.length].uninitSlice!T;
+
+        auto m = (allowDestroy && matrix._stride!1 == 1) ? matrix.assumeCanonical : matrix.as!T.slice.canonical;
+
+        static if(is(T == double) || is(T == float))
+            orgqr!T(m, tau, work);
+        else
+            ungqr!T(m, tau, work);
+        return m.transposed;
+    }
+    
+    /++
+    Extract the R matrix
+    +/
+    auto R()
+    {
+        auto r = uninitSlice!T(matrix.shape).universal;
```

Comment:
```
import mir.algorithm.iteration: eachLower;
auto r = matrix.transposed.slice;  
r.eachLower!"a = 0";
return r; // or r.universal depending on API you like to have
```

Reply:
hope this will help to fix travis

Reply:
This PR is also behind one commit from whatever you fixed. That may also be a factor.

Reply:
There will need to be some additional work to get this working with rectangular matrices also. I've noticed some issues already trying to compare the results with what I get in R.